opentelemetry/benches/context_attach.rs

```diff
@@ -0,0 +1,119 @@
+use criterion::{
+    black_box, criterion_group, criterion_main, measurement::WallTime, BenchmarkGroup, BenchmarkId,
+    Criterion, Throughput,
+};
+use opentelemetry::{
+    trace::{SpanContext, TraceContextExt},
+    Context,
+};
+
+// Run this benchmark with:
+// cargo bench --bench context_attach
+
+fn criterion_benchmark(c: &mut Criterion) {
+    let span_context = Context::new().with_remote_span_context(SpanContext::empty_context());
+    let contexts = vec![
+        ("empty_cx", Context::new()),
+        ("single_value_cx", Context::new().with_value(Value(4711))),
+        ("span_cx", span_context),
+    ];
+    for (name, cx) in contexts {
+        single_cx_scope(&mut group(c), name, &cx);
+        nested_cx_scope(&mut group(c), name, &cx);
+        overlapping_cx_scope(&mut group(c), name, &cx);
+    }
+}
+
+fn single_cx_scope(
+    group: &mut BenchmarkGroup<'_, WallTime>,
+    context_type: &str,
+    context: &Context,
+) {
+    let _restore = Context::current().attach();
+    group.throughput(Throughput::Elements(1)).bench_function(
+        BenchmarkId::new("single_cx_scope", context_type),
+        |b| {
+            b.iter_batched(
+                || context.clone(),
+                |cx| {
+                    single_cx(cx);
+                },
+                criterion::BatchSize::SmallInput,
+            );
+        },
+    );
+}
+
+#[inline(never)]
+fn single_cx(cx: Context) {
+    let _ = black_box(cx.attach());
+    let _ = black_box(dummy_work());
+}
+
+fn nested_cx_scope(group: &mut BenchmarkGroup<'_, WallTime>, cx_type: &str, context: &Context) {
+    let _restore = Context::current().attach();
+    group.throughput(Throughput::Elements(1)).bench_function(
+        BenchmarkId::new("nested_cx_scope", cx_type),
+        |b| {
+            b.iter_batched(
+                || (context.clone(), context.clone()),
+                |(cx1, cx2)| {
+                    nested_cx(cx1, cx2);
+                },
+                criterion::BatchSize::SmallInput,
+            );
+        },
+    );
+}
+
+#[inline(never)]
+fn nested_cx(cx1: Context, cx2: Context) {
+    let _ = black_box(cx1.attach());
+    let _ = black_box(cx2.attach());
```

Comment:
I don't this nest contexts. Binding a value to `_` immediately drops it ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=49ecb5881848e0ae967ce63a00da371c)) , to the opposite of binding to an unused variable

```suggestion
    let _cx1 = black_box(cx1.attach());
    let _cx2 = black_box(cx2.attach());
```

Reply:
https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry/src/context.rs#L262-L264

Reply:
Thanks for finding this. I'll make sure to update the benchmark.